packages/common/src/content/index.ts

```diff
@@ -737,15 +738,145 @@ export const setContentType = (
   // get family and normalized type based on new type
   const normalizedType = normalizeItemType({ ...content.item, type });
   const family = getFamily(normalizedType);
+  const contentTypeIcon = setContentTypeIcon(normalizedType);
+  const contentTypeLabel = setContentTypeLabel(
+    normalizedType,
+    content.isProxied
+  );
   // TODO: don't we need to update content.item.type too?
-  const updated = { ...content, type, family, normalizedType };
+  const updated = {
+    ...content,
+    type,
+    family,
+    normalizedType,
+    contentTypeIcon,
+    contentTypeLabel,
+  };
   // update the relative URL to the content
   // which is based on type and family
   return appendContentUrls(updated, {
     relative: getContentRelativeUrl(updated),
   });
 };
 
+/**
+ * @param normalizedType
+ * @returns content type icon based on the normalizedType
+ */
```

Comment:
Description of what this does should be the first line of the comment. 
You can add params details like this `@param <paramname> <details>`
The @returns type will automatically be displayed based on the return type of the function, but you can provide additional details